docs: improve environment variables section (#1124)
apify/apify-docs: sources/platform/actors/development/programming_interface/environment_variables.md

## System environment variables

Apify sets several system environment variables for each Actor run. These variables provide essential context and information about the Actor's execution environment.

Here's a table of key system environment variables:

| Environment Variable | Description |
|----------------------|-------------|
| `ACTOR_ID` | ID of the Actor. |
| `ACTOR_RUN_ID` | ID of the Actor run. |
| `ACTOR_BUILD_ID` | ID of the Actor build used in the run. |
| `ACTOR_BUILD_NUMBER` | Build number of the Actor build used in the run. |
| `ACTOR_TASK_ID` | ID of the Actor task. Empty if Actor is run outside of any task, e.g. directly using the API. |

> | `ACTOR_EVENTS_WEBSOCKET_URL` | Websocket URL where Actor may listen for events from Actor platform. |

Reviewer: A link to the events section would be nice here

| `ACTOR_DEFAULT_DATASET_ID` | ID of the dataset where you can push the data. |
| `ACTOR_DEFAULT_KEY_VALUE_STORE_ID` | ID of the key-value store where the Actor's input and output data are stored. |

> | `ACTOR_DEFAULT_REQUEST_QUEUE_ID` | ID of the request queue that stores and handles requests that you enqueue. |

Reviewer: Well, yeah, but you can push to any dataset/whatever in your account based on name or ID. This should mention that these are created specifically for a single run.

Author: rephrased description to mention that these are run specific values

> | `ACTOR_INPUT_KEY` | Key of the record in the default key-value store that holds the Actor input. Typically **INPUT**, but may vary. |

Reviewer: No need to mention the default value here. A link explaining what an input actually is would be helpful.

Author: added link to input section of docs

| `ACTOR_MAX_PAID_DATASET_ITEMS` | For paid-per-result Actors, the user-set limit on returned results. Do not exceed this limit. |
| `APIFY_HEADLESS` | If **1**, web browsers inside the Actor should run in headless mode (no windowing system available). |

> | `APIFY_IS_AT_HOME` | Set to **1** if the Actor is running on Apify servers. |

Reviewer: ```suggestion
| `APIFY_IS_AT_HOME` | Contains **1** if the Actor is running on Apify servers. |
```

> | `ACTOR_MEMORY_MBYTES` | Size of memory allocated for the Actor run, in megabytes. Used to optimize memory usage. |

Reviewer: I'd rather say "Can be used to optimize memory usage or finetuning of low-level external libraries". There's not much that your average actor can do with this information, except maybe for adjusting the size of some cache.

> | `APIFY_PROXY_PASSWORD` | The Apify Proxy password of the user who started the Actor. |

Reviewer: ```suggestion
| `APIFY_PROXY_PASSWORD` | A password that can be used to use the Apify Proxy on behalf of the user who started the Actor. |
```

> | `APIFY_PROXY_PORT` |  |

Reviewer: ```suggestion
| `APIFY_PROXY_PORT` | TCP port number to be used for connecting to the Apify Proxy  |
```

> | `APIFY_PROXY_STATUS_URL` |  |

Reviewer: ```suggestion
| `APIFY_PROXY_STATUS_URL` | URL of a page that contains information about the proxy used to access it. Machine-readable output may be obtained by appending `?format=json` to the URL. |
```

> | `ACTOR_STANDBY_PORT` | TCP port for the Actor to start an HTTP server to receive messages from Actor Standby. |

Reviewer: ```suggestion
| `ACTOR_STANDBY_PORT` | TCP port for the Actor to start an HTTP server on to receive messages in the Actor Standby mode. |
```

Also, link to Actor standby docs would be nice here :slightly_smiling_face:

| `ACTOR_STARTED_AT` | Date when the Actor was started. |
| `ACTOR_TIMEOUT_AT` | Date when the Actor will time out. |
| `APIFY_TOKEN` | API token of the user who started the Actor. |
| `APIFY_USER_ID` | ID of the user who started the Actor. May differ from the Actor owner. |

> | `ACTOR_WEB_SERVER_PORT` | TCP port for the Actor to start an HTTP server to receive external messages. |

Reviewer: ```suggestion
| `ACTOR_WEB_SERVER_PORT` | TCP port for the Actor to start an HTTP server on to receive external messages or expose a monitoring or control UI. |
```

| `ACTOR_WEB_SERVER_URL` | Unique public URL for accessing the Actor run web server from the outside world. |

> | `APIFY_API_PUBLIC_BASE_URL` | Public URL of the API, typically set to api.apify.com |

Reviewer: ```suggestion
| `APIFY_API_PUBLIC_BASE_URL` | Public URL of the Apify API, which may be used to interact with the platform. Typically set to api.apify.com |
```

| `APIFY_DEDICATED_CPUS` | Number of CPU cores reserved for the actor, based on allocated memory. |

> | `APIFY_DISABLE_OUTDATED_WARNING` |  |

Reviewer: ```suggestion
| `APIFY_DISABLE_OUTDATED_WARNING` | If you manually set this to `1`, the Apify SDK will not warn you if there is a new version available |
```

Author: Is there a default value here? I assume `0`?

Reviewer: It's an empty string by default

Author: I guess then we don't have to mention default value if it is empty string

> | `APIFY_FACT` |  |

Reviewer: ```suggestion
| `APIFY_FACT` | Easter egg |
```

We may just leave this one out I guess :slightly_smiling_face:

Author: Best to remove that if nothing of value is added :D

Reviewer: Yeah, I think that if somebody can't understand it's an easter egg when they encounter it, they're lost anyway.

| `APIFY_WORKFLOW_KEY` | Identifier used for grouping related runs and API calls together. |

> | `APIFY_META_ORIGIN` |  |

Reviewer: This basically explains how the run was started... possible values here:

![image](https://github.com/user-attachments/assets/0b06d696-3592-4e66-b923-6e47ad3cf4f2)

Author: added link with origin section

> | `APIFY_SDK_LATEST_VERSION` |  |

Reviewer: ```suggestion
| `APIFY_SDK_LATEST_VERSION` | Latest available version of the Apify SDK for Javascript, used for checking for updates |
```

<!-- vale Microsoft.RangeFormat = NO -->

:::note Date format

All date-related variables use the UTC timezone and are in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format (e.g., _2022-07-13T14:23:37.281Z_).

:::
<!-- vale Microsoft.RangeFormat = YES -->

## Access environment variables

You can access environment variables in your code as follows:

<Tabs groupId="main">
<TabItem value="JavaScript" label="JavaScript">

In Node.js, use the `process.env` object:

```js
console.log(process.env.APIFY_USER_ID);
```

</TabItem>
<TabItem value="Python" label="Python">

In Python, use the `os.environ` dictionary:

```python
import os
print(os.environ['APIFY_USER_ID'])
```

</TabItem>
</Tabs>

## Use the `Configuration` class

> For more convenient access to Actor configuration, use the `Configuration` class

Reviewer: The configuration should be a link here https://docs.apify.com/sdk/js/reference/class/Configuration

| `ACTOR_DEFAULT_KEY_VALUE_STORE_ID` | ID of the key-value store where the Actor's input and output data are stored. |
| `ACTOR_DEFAULT_REQUEST_QUEUE_ID` | ID of the request queue that stores and handles requests that you enqueue. |
| `ACTOR_INPUT_KEY` | Key of the record in the default key-value store that holds the Actor input. Typically **INPUT**, but may vary. |
| `ACTOR_MAX_PAID_DATASET_ITEMS` | For paid-per-result Actors, the user-set limit on returned results. Do not exceed this limit. |
| `APIFY_HEADLESS` | If **1**, web browsers inside the Actor should run in headless mode (no windowing system available). |
| `APIFY_IS_AT_HOME` | Contains **1** if the Actor is running on Apify servers. |
| `ACTOR_MEMORY_MBYTES` | Size of memory allocated for the Actor run, in megabytes. Can be used to optimize memory usage or finetuning of low-level external libraries. |
| `APIFY_PROXY_PASSWORD` | Password for accessing Apify Proxy services. This password enables the Actor to utilize nproxy servers on behalf of the user who initiated the Actor run. |
| `APIFY_PROXY_PORT` | TCP port number to be used for connecting to the Apify Proxy. |

> | `APIFY_PROXY_STATUS_URL` | Endpoint URL for regrieving proxy status information. Appending `?format=json` to this URL returns the data in JSON format for programmatic processing. |

Reviewer: ```suggestion
| `APIFY_PROXY_STATUS_URL` | URL for retrieving proxy status information. Appending `?format=json` to this URL returns the data in JSON format for programmatic processing. |
```

| `ACTOR_STANDBY_PORT` | TCP port for the Actor to start an HTTP server to receive messages in the [Actor Standby](/platform/actors/development/programming-interface/standby) mode. |
| `ACTOR_STARTED_AT` | Date when the Actor was started. |
| `ACTOR_TIMEOUT_AT` | Date when the Actor will time out. |
| `APIFY_TOKEN` | API token of the user who started the Actor. |
| `APIFY_USER_ID` | ID of the user who started the Actor. May differ from the Actor owner. |
| `ACTOR_WEB_SERVER_PORT` | TCP port for the Actor to start an HTTP server on. This server can be used to receive external messages or expose monitoring and control interfaces. |
| `ACTOR_WEB_SERVER_URL` | Unique public URL for accessing the Actor run web server from the outside world. |
| `APIFY_API_PUBLIC_BASE_URL` | Public URL of the Apify API. May be used to interact with the platform programmatically. Typically set to `api.apify.com`. |
| `APIFY_DEDICATED_CPUS` | Number of CPU cores reserved for the actor, based on allocated memory. |

> | `APIFY_DISABLE_OUTDATED_WARNING` | Controls the display of outdated version warnings. Set to `1` to suppress notifications about updates. Default value `0`. |

Reviewer: ```suggestion
| `APIFY_DISABLE_OUTDATED_WARNING` | Controls the display of outdated version warnings. Set to `1` to suppress notifications about updates. |
```

> | `AP IFY_WORKFLOW_KEY` | Identifier used for grouping related runs and API calls together. |

Reviewer: ```suggestion
| `APIFY_WORKFLOW_KEY` | Identifier used for grouping related runs and API calls together. |
```

| `APIFY_META_ORIGIN` | Specifies how an Actor run was started. |
| `APIFY_SDK_LATEST_VERSION` | Specifies the most recent release version of the Apify SDK for Javascript. Used for checking for updates. |

> | `APIFY_INPUT_SECRETS_KEY_FILE` |  |
> | `APIFY_INPUT_SECRETS_KEY_PASSPHRASE` |  |

Reviewer: ```suggestion
| `APIFY_INPUT_SECRETS_KEY_FILE` | Path to the secret key that may be used to decrypt [Secret inputs](/platform/actors/development/actor-definition/input-schema/secret-input) |
| `APIFY_INPUT_SECRETS_KEY_PASSPHRASE` | Passphrase for the input secret key mentioned above |
```
